topydo/cli/UILoader.py

```diff
@@ -0,0 +1,34 @@
+# Topydo - A todo.txt client written in Python.
+# Copyright (C) 2014 - 2015 Bram Schoenmakers <me@bramschoenmakers.nl>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Entry file for the Python todo.txt CLI. """
+
+import sys
+from topydo.cli.CLI import CLIApplication
+from topydo.cli.Prompt import PromptApplication
+
+def main():
+    """ Main entry point of the CLI. """
+    try:
+        if sys.argv[1] == 'prompt':
```

Comment:
Or maybe create some function inside `UILoader` like `_preprocess_flags` and do there something like:

```python
def _preprocess_flags():
    args = sys.argv[1:]

    if PY2:
        args = [arg.decode('utf-8') for arg in args]

    try:
        opts, args = getopt.getopt(args, "c:d:ht:v")
    except getopt.GetoptError as e:
        error(str(e))
        sys.exit(1)

    return (opts, args)
```

and in `UILoader.main` do something like this:

```python
args = _preprocess_flags()

if args[1][0] == 'prompt':
    PromptApplication(args).run()
else:
    CLIApplication(args).run()
```

Then of course `_process_flags` would operate on args passed to objects as above, instead of on `sys.argv`.